### DISCUSSION TITLE
希望动态过滤功能能新增屏蔽广告动态（例如“up主的推荐”）

### DISCUSSION THREAD

**Discussion by WerAndRingo** [the1812/Bilibili-Evolved#4425] [功能建议 / Ideas]
希望能屏蔽带有up主推荐商品的动态，见图
图片链接：https://s2.loli.net/2024/02/05/X6jfwGtxbD3TeZC.png

**Comment by miaonai** [comment]
https://greasyfork.org/zh-CN/scripts/475122
这个可以暂时替代，修改关键词不是很方便，需要自行在Tampermonkey里编辑关键词
刷出新的动态需要屏蔽会有延迟，默认每5秒屏蔽一次，可以自行修改

**Comment by zhongfly** [comment]
我自己解决了
在ublock等广告屏蔽插件的自定义静态规则中，添加以下2条规则
```
t.bilibili.com##div.bili-dyn-list__item:has(span[data-type="goods"])
space.bilibili.com##div.bili-dyn-list__item:has(span[data-type="goods"])
```
可以在自己的动态主页和别人的动态页中屏蔽带有商品推荐的动态

![image](https://github.com/the1812/Bilibili-Evolved/assets/11155705/6b4ed667-9ec1-4ec3-ba2b-d0c2ec8027b1)
注：图中的规则多了`[data-type="2"]`是为了只过滤图文动态，提高效率，但是没登录时会失效。

**Comment by qtqz** [comment]
咱是把常见推广品名和文案全加到动态过滤了
